server/routes/avatar/utils.js

```diff
@@ -0,0 +1,99 @@
+import sharp from 'sharp';
+import { throttle } from 'underscore';
+
+import { Cookies } from 'meteor/ostrio:cookies';
+
+import { Users } from '../../../app/models/server';
+import { getAvatarColor } from '../../../app/utils';
+import { settings } from '../../../app/settings/server';
+
+const FALLBACK_LAST_MODIFIED = 'Thu, 01 Jan 2015 00:00:00 GMT';
+
+const cookie = new Cookies();
+
+export const getCacheTime = (req) => req.query.cacheTime || settings.get('Accounts_AvatarCacheTime');
+
+export const serveAvatar = (avatar, req, res) => {
```

Comment:
I don't think these functions should receive the express objects (`req`, `res`), I think that the express objects, should only be at the level they are received, and not be passed to levels below. :thinking:

Reply:
Maybe the function should only be moved to another place instead? Because by the name `serverAvatar` I guess it is *theoretically* in the same layer as the route handlers 🤔

Reply:
yep, I guess so.. fixed all other comments btw =)